cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -62,6 +62,7 @@
 from charms.layer.kubernetes_common import kubeproxyconfig_path
 from charms.layer.kubernetes_common import configure_kube_proxy
 from charms.layer.kubernetes_common import get_version
+from charms.layer.kubernetes_common import manage_docker_opts, manage_registry_certs
```

Comment:
I'm all for common functions, but are these actually expected to be used by something else besides kubernetes-worker?

Reply:
@hyperbolic2346 not today, but I think they will one day.  Maybe our masters start running kubelets for pods with a special label.  Maybe we move our LB to an nginx container with the image hosted in a private registry. Maybe we have a k8s-monitor charm for in-cluster prometheus and fluentd pods.

More than future potential usefulness, i put these funcs here because they feel like lib funcs to me.  They're not reactive, and we don't have a `lib/charms/layer` home for k8s-worker.  k8s-common felt like a good place to me, but i'm happy to entertain alternate homes.

Reply:
> Maybe our masters start running kubelets for pods with a special label.

I'd say this one in particular has been on our wishlist for a while. So yes, there is potential for this to be common code in the future.

But personally, I wouldn't put anything in layer-kubernetes-common unless it's actively being used by multiple charms, or there are immediate plans to do so. It's a bit easier to move stuff from worker to common later than it is to go the other direction.

> More than future potential usefulness, i put these funcs here because they feel like lib funcs to me. They're not reactive, and we don't have a lib/charms/layer home for k8s-worker.

Feel free to make one! I guess that would be `lib/charms/layer/kubernetes_worker.py`?

In any case, seeing as I just let a similar non-common thing go into layer-kubernetes-common [here](https://github.com/juju-solutions/kubernetes/pull/191#discussion_r236454471), I would be fine with this too. @kwmonroe I'd encourage you to just put these functions wherever feels right to you.